src/DataStructures/DataBox/Deferred.hpp

```diff
@@ -202,12 +262,31 @@ class deferred_assoc_state : public assoc_state<Rt> {
  * \example
  * Construction of a Deferred with an object followed by mutation:
  * \snippet Test_Deferred.cpp deferred_with_update
+ *
+ * \warning If passed a lazy function that returns a `const T& t` (a const
+ * lvalue reference) the underlying data stored is actually a pointer to `t`,
+ * which will be dereferenced upon retrieval. This could lead to a dangling
+ * pointer/reference if care isn't taken. The reason for this design is that
+ * 1. We need to be able to retrieve data stored in the ConstGlobalCache from
+ *    the DataBox. The way to do this is to have a pointer to the
+ *    ConstGlobalCache inside the DataBox alongside compute items that return
+ *    `const T&` to the ConstGlobalCache data.
+ * 2. Functions used in the DataBox shouldn't return references, they are
+ *    compute tags and should return by value. If we find an actual use-case for
+ *    compute tags returning `const T&` where referencing behavior is undesired
+ *    then this design needs to be reconsidered. It is currently the least
+ *    breaking way to implement the desired functionality.
```

Comment:
Actually state what "the desired functionality" is for clarity. Can you just refer back the previous item in this very list?